File: SistemaParadaCerta/APIs-ParadaCerta/paradacerta-api/src/main/java/com/paradacerta/api/model/SessaoStatus.java
package com.paradacerta.api.model;

public enum SessaoStatus {
    AGUARDANDO_CONFIRMACAO,
    EM_USO,
    ATIVA,
    ENCERRADA,
    CANCELADA;

    public boolean isAtivoOuAguardando() {
        return this == AGUARDANDO_CONFIRMACAO || this == EM_USO || this == ATIVA;
    }
}

File: SistemaParadaCerta/AplicacaoMobile-ParadaCerta/ParadaCerta/app/src/main/java/com/example/paradacerta/models/SessaoStatus.kt
package com.example.paradacerta.models

/**
 * Status da sessão/reserva — espelha o enum SessaoStatus do backend.
 * Builds antigos do app podem receber valores ATIVA/ENCERRADA/CANCELADA;
 * builds novos também recebem AGUARDANDO_CONFIRMACAO e EM_USO.
 */
enum class SessaoStatus {
    AGUARDANDO_CONFIRMACAO,
    EM_USO,
    ATIVA,
    ENCERRADA,
    CANCELADA;

    companion object {
        fun fromString(raw: String?): SessaoStatus {
            if (raw.isNullOrBlank()) return ATIVA
            return runCatching { valueOf(raw.uppercase()) }.getOrDefault(ATIVA)
        }
    }
}
